UniStuttgart-VISUS/megamol
# Animation Editor

Since the Cinematic View is a bit dated and very restricted WRT what you can animate, we need a new approach.

## Summary of Changes
- Animation Editor at GUI/Frontend level
- VolumeToTable
- Cam Poses as general util
- Drift velocity in ParticleThermodyn
- Profiling logging can be set from Lua
- Profiling logging can be paused
- GUI windows can request resources
- Graph layouting command

## Fixes
- SplitView crashes when TimeLord is active
- ArrowRenderer light direction
- AdiosToTable performance
- -e cmdline parsing
- Profiling fix when configuring at runtime
- ADIOSFlexVolume updates correctly

TODOs:
- [x] Mouse Zoom
- [x] Animation Playback (Scrubbing works)
- [x] Movie/Screenshot output
- [x] Capture/Interpolate a camera path
- [x] Tangents on subdivision
- [ ] Finish up Readme
- [ ] Frame index inconsistency with Screenshooter
- [ ] Add key in context menu (?) of all parameters
- [ ] Zoom limiters for Animation Editor

## Test Instructions
Try out features like auto capture and graph write-back, see [frontend/services/gui/README.md](https://github.com/UniStuttgart-VISUS/megamol/blob/bbbdf73a39e374e33f42577ca861c1f3f02761c9/frontend/services/gui/README.md#animation-editor)

Example of an animated FloatParam
![megamol_screenshot_1](https://user-images.githubusercontent.com/26002546/205004478-a137e997-44c9-46e7-b0e1-2878b492cc72.png)
